contrib/completion/git-completion.bash

```diff
@@ -2986,24 +2986,56 @@ _git_show_branch ()
 	__git_complete_revlist
 }
 
+__gitcomp_directories ()
+{
+	local _tmp_dir _tmp_completions
+
+	# Get the directory of the current token; this differs from dirname
+	# in that it keeps up to the final trailing slash.  If no slash found
+	# that's fine too.
+	[[ "$cur" =~ .*/ ]]
+	_tmp_dir=$BASH_REMATCH
+
+	# Find possible directory completions, adding trailing '/' characters,
+	# de-quoting, and handling unusual characters.
+	_tmp_completions=()
+	while IFS= read -r -d $'\0' c ; do
+		_tmp_completions+=("$c/")
+	done < <(git ls-tree -z -d --name-only HEAD $_tmp_dir)
+
+	if [[ ${#_tmp_completions[@]} != 0 ]]; then
+		# There were some directory completions, so find ones that
+		# start with "$cur", the current token, and put those in COMPREPLY
+		for c in "${_tmp_completions[@]}"; do
+			if [[ $c == "$cur"* ]]; then
+				COMPREPLY+=("$c")
```

Comment:
I think we can merge these two loops:

If we replace `_tmp_completions=()` above with `local found=`, and then insert a `found=t` here, we can move this conditional into the `while IFS= [...]` loop and get rid of the `for c in [...]` loop as well as get rid of the `_tmp_completions` array.

The `elif [[ "$cur" =~ /$ ]]` below would then also need to add the condition `test -z "$found"`.